### ISSUE TITLE
Proxy annex configs from .datalad/config to annex "runtime"

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#3456]
git (annex) does not pick up `.datalad/config` as a configuration file to be considered for its operations.  But such options as `annex.security.allowed-*-addresses` might need to be "committed" into the dataset's config to enable access to some addresses.  See e.g.  [redirection to ftp](http://git-annex.branchable.com/bugs/regression__58___http_downloads_redirecting_to_ftp_are_no_longer_supported/#comment-82edbccee752ca3876437d7c034e6461) issue which was brought up by #3321 .

The easiest solution would be to instruct git (annex) to **also** consider  .datalad/config, but AFAIK it is not possible. `man git-config` says that `GIT_CONFIG` environment variable overrides the default `.git/config` (or `~/.gitconfig` in case of `--global`).  

So another solution would be to load dataset specific config and for each invocation of git (annex) pass explicitly the options which are not otherwise defined in the config.

May be there is a better solution?

**Comment by kyleam** [comment]
Nothing better is coming to my mind.

**Comment by mih** [comment]
How would we ensure that nobody unknowingly pulls security related config setting via a dataset update?

**Comment by yarikoptic** [comment]
Yeah , hooks and configs coming from a remote are opening security concerns. Probably we should sway away from that in general .

May be though there could be a set of config settings which we, upon interactive user approval, should duplicate into local .git/config ? (Like those security settings mentioned in original description)

**Comment by bpoldrack** [comment]
Given how the issue is, I think there aren't a lot of use cases. The original one is sort of valid, but the wrong approach in my mind. Setting such a config locally could/should be a matter of a postclone routine for specific cases.

> upon interactive user approval

Doesn't scale. I think the most likely case where one would want to allow such things is when the workflow is running in a somewhat confined environment. Say one has stuff on a institute's infrastructure and clones (for running analysis or sth) or done within that network where you trust there aren't arbitrary commits from the outside world. This will likely be scripted jobs where interactive query is not an option (same would be true for our CI runs).

I think, we should not auto-pull security related configs ever. A dataset that comes with this requirement, should be documented that way and could additionally provide a procedure setting them for convenience. Training users to be unaware of security implications by automating this or by training them to always answer a question with "yes" is bad.
